pkg/vex/vex.go

```diff
@@ -104,3 +106,68 @@ func decodeCSAF(r io.ReadSeeker) (VEX, error) {
 	}
 	return newCSAF(adv), nil
 }
+
+type NotAffected func(vuln types.DetectedVulnerability, product, subComponent *core.Component) (types.ModifiedFinding, bool)
+
+func filterVulnerabilities(result *types.Result, bom *core.BOM, fn NotAffected) {
+	components := lo.MapEntries(bom.Components(), func(id uuid.UUID, component *core.Component) (string, *core.Component) {
+		return component.PkgIdentifier.UID, component
+	})
+
+	result.Vulnerabilities = lo.Filter(result.Vulnerabilities, func(vuln types.DetectedVulnerability, _ int) bool {
+		if vuln.PkgIdentifier.PURL == nil {
+			return true
+		}
+
+		c, ok := components[vuln.PkgIdentifier.UID]
+		if !ok {
+			return true // Should never reach here
```

Comment:
I suggest adding log anyway if this happens.

Reply:
https://github.com/aquasecurity/trivy/pull/6728/commits/34483c233d1bc8df08802b2749220f997209aefe